File: quark/DatawireFS_impl.js
(function () {
    "use strict";

    var os = require('os');
    var fs = require('fs');

    var quark = require('quark').quark;

    var DatawireFS = (function () {
        function DatawireFS() {}

        DatawireFS.userHomeDir = function () {
            return os.homedir();
        };

        DatawireFS.fileContents = function (path) {
            try {
                return fs.readFileSync(path, 'utf-8');
            }
            catch (e) {
                var runtime = quark.concurrent.Context.runtime();
                runtime.fail("failure reading " + path + ": " + e)
            }
        };

        return DatawireFS;
    })();

    module.exports = DatawireFS;
})();

File: quark/DatawireFS_impl.py
__version__ = '1.0.0';

import os

import quark

__all__ = """DatawireFS""".split(' ')

class DatawireFS (object):
    @classmethod
    def userHomeDir(klass):
        return os.path.expanduser('~')

    @classmethod
    def fileContents(klass, path):
        try:
            inputFile = open(path, "r")
            return inputFile.read()
        except Exception as e:
            runtime = quark.concurrent.Context.runtime()
            runtime.fail("failure reading %s: %s" % (path, e))
